src/Items/ItemChest.h

```diff
@@ -43,36 +45,37 @@ class cItemChestHandler:
 		// Check if the block ignores build collision (water, grass etc.):
 		BLOCKTYPE clickedBlock;
 		NIBBLETYPE clickedBlockMeta;
-		Vector3i blockPos(a_BlockX, a_BlockY, a_BlockZ);
-		a_World.GetBlockTypeMeta(a_BlockX, a_BlockY, a_BlockZ, clickedBlock, clickedBlockMeta);
+		a_World.GetBlockTypeMeta(absPos.x, absPos.y, absPos.z, clickedBlock, clickedBlockMeta);
+
 		cChunkInterface ChunkInterface(a_World.GetChunkMap());
-		auto blockHandler = BlockHandler(clickedBlock);
-		if (blockHandler->DoesIgnoreBuildCollision(ChunkInterface, blockPos, a_Player, clickedBlockMeta))
+		// Check if the block ignores build collision (water, grass etc.):
+		if (BlockHandler(clickedBlock)->DoesIgnoreBuildCollision(ChunkInterface, absPos, a_Player, clickedBlockMeta))
 		{
-			blockHandler->OnPlayerBreakingBlock(ChunkInterface, a_World, a_Player, blockPos);
+			a_World.DropBlockAsPickups(absPos, &a_Player, nullptr);
```

Comment:
Do these "replaceable" blocks ever drop anything when replaced?
Also why this change?

Reply:
Think of grass, seeds will be dropped sometimes.

And this was changed to use the new DropBlockAsPickups to enable this functionality, instead of just removing it without dropping anything.

Reply:
I don't recall them popping anything out when say, putting a torch in them, only when left clicking to break.

Reply:
Admittedly I haven't tested. But they _should_. And in any case it should be consistent between chests and any other item.

Reply:
According to https://minecraft.fandom.com/wiki/Grass#Trivia grass does not drop seed when its being replacing.